x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/shared/SharedBytes.java

```diff
@@ -212,20 +246,37 @@ public static void copyToCacheFileAligned(
      * @param fileChannelPos position in {@code fc} to write to
      * @param progressUpdater callback to invoke with the number of copied bytes as they are copied
      * @param buffer bytebuffer to use for writing
+     * @param metricsConsumer a consumer that will be provided with the read/write metrics upon completion of the copy
      * @return the number of bytes copied
      * @throws IOException on failure
      */
-    public static int copyToCacheFileAligned(IO fc, InputStream input, int fileChannelPos, IntConsumer progressUpdater, ByteBuffer buffer)
-        throws IOException {
+    public static int copyToCacheFileAligned(
+        IO fc,
+        InputStream input,
+        int fileChannelPos,
+        IntConsumer progressUpdater,
+        ByteBuffer buffer,
+        CacheCopyMetricsConsumer metricsConsumer
```

Comment:
I assume this is where we can choose to pass down the context for things like warming or not, shardId, etc

Reply:
Yes, please see https://github.com/elastic/elasticsearch-serverless/pull/2525 for examples of its use